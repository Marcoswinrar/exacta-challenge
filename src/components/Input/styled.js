import InputMask from 'react-input-mask'
import styled from 'styled-components'
import media from 'styled-media-query'
import { UserFormFieldStyle } from '../../config/styles'

export const InputWrapper = styled.div`
  display: flex;
  flex-direction: column;
  
  ${media.lessThan("large")`
    width: 100%;
  `}

  .invalid {
    border-color: red;
  }

`
export const Input = styled(InputMask)`
  ${UserFormFieldStyle} 

  :focus {
    outline: none;
    border-color: blue;
  }

`
export const Label = styled.label`
  font-size: .8rem;
  text-transform: uppercase;
  font-weight: bolder;
`

export const Separator = styled.span`
  width: 4px;
  height: 2px;
  background-color: black;
  margin: auto 3px 15px 3px;

  ${media.lessThan("large")`
    height: 0;
  `}
`