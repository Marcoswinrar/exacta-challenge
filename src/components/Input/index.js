import styled from 'styled-components'
import { UserFormFieldStyle } from '../../config/styles'

export const Input = styled.input`
  ${UserFormFieldStyle} 

`
export const InputLabel = styled.label`
  font-size: .8rem;
  text-transform: uppercase;
  font-weight: bolder;
`