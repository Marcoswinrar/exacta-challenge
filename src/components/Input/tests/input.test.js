import { fireEvent, render, screen } from "@testing-library/react"
import { act } from "react-dom/test-utils"
import Input from ".."

describe("Componente <Input />", () => {

  it("Deve renderizar o componente sem problemas", () => {
    const { container } = render(
      <Input
        label="Input"
        type="text"
        value="Teste"
        hasSeparator
      />
    )

    expect(container).toMatchSnapshot()
    expect(screen.queryByText(/Input/)).toBeInTheDocument()
  })

  it("Deve renderizar o Input com mÃ¡scara caso seja passado ao componente", () => {
    render(
      <Input
        label="Input"
        type="text"
        value="Teste"
        mask="123456-0"
      />
    )

    expect(screen.getByDisplayValue(/123456-0/)).toBeInTheDocument()
  })

  it("Deve chamar o onChange e  conter o valor correto ao digitar no campo de Input", () => {
    const onChange = jest.fn()

    render(
      <Input
        label="Input"
        type="text"
        onChange={onChange}
      />
    )

    const input = screen.getByLabelText(/Input/)

    act(() => fireEvent.change(input, { target: { value: 'Teste1234560' } }))

    expect(onChange).toBeCalledTimes(1)
    expect(screen.getByDisplayValue(/Teste1234560/)).toBeInTheDocument()

  })
})