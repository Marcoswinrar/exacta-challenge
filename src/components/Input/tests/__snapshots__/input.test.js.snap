// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente <Input /> Deve renderizar o componente sem problemas 1`] = `
<div>
  <div
    class="sc-bczRLJ OEsGh"
  >
    <label
      class="sc-dkzDqf hNEPNr"
      for="Input"
    >
      Input
    </label>
    <input
      class="sc-gsnTZi ifYwBn"
      focused="false"
      id="Input"
      type="text"
      value="Teste"
    />
  </div>
  <span
    class="sc-hKMtZM gTAjIp"
  />
</div>
`;
